src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs

```diff
@@ -57,14 +179,18 @@ public Task Save(IContainSagaData sagaData, SagaCorrelationProperty correlationP
                 Put = new Put
                 {
                     Item = Serialize(sagaData, 0),
-                    ConditionExpression = "attribute_not_exists(#SK)", //Fail if already exists.
+                    ConditionExpression = "attribute_not_exists(#version)", // fail if a saga (not just the lock) already exists
```

Comment:
verify: this might overwrite an existing lock in case we try to save concurrently without getting a lock

sidenote: the comment doesn't make sense if pessimistic locking isn't enabled.